chardet/charsetprober.py

```diff
@@ -26,70 +26,120 @@
-            international character. all contiguous sequences of markers are
-            replaced by a single space ascii character.
+        We define three types of bytes:
+        alphabet: english alphabets [a-zA-Z]
+        international: international characters [\x80-\xFF]
+        marker: everything else [^a-zA-Z\x80-\xFF]
+
+        The input buffer can be thought to contain a series of words delimited
+        by markers. This function works to filter all words that contain at
+        least one international character. All contiguous sequences of markers
+        are replaced by a single space ascii character.
+
+        This filter applies to all scripts which do not use English characters.
         """
         filtered = BytesIO()
 
-        # this regex expression filters out only words that have at-least
-        # one international character. the word may include one marker
-        # character at the end
-        words = \
-            re.findall(b'[a-zA-Z]*[\x80-\xFF]+[a-zA-Z]*[^a-zA-Z\x80-\xFF]?',
-                       buf)
+        # This regex expression filters out only words that have at-least one
+        # international character. The word may include one marker character at
+        # the end.
+        words = re.findall(
+            b'[a-zA-Z]*[\x80-\xFF]+[a-zA-Z]*[^a-zA-Z\x80-\xFF]?', buf)
 
         for word in words:
             filtered.write(word[:-1])
 
-            # if the last character in the word is a marker, replace it with a
+            # If the last character in the word is a marker, replace it with a
             # space as markers shouldn't affect our analysis (they are used
             # similarly across all languages and may thus have similar
-            # frequencies)
+            # frequencies).
             last_char = word[-1:]
-            last_char = last_char if last_char.isalpha() or \
-                last_char >= b'\x80' else b' '
+            last_char = last_char if (last_char.isalpha() or
+                                      last_char >= b'\x80') else b' '
             filtered.write(last_char)
 
         return filtered.getvalue()
 
-    def filter_with_english_letters(self, aBuf):
-        # TODO
-        return aBuf
+    @staticmethod
+    def filter_with_english_letters(buf):
+        """
+        Returns a copy of ``buf`` that retains only the sequences of English
+        alphabet and high byte characters that are not between <> characters.
+        Also retains English alphabet and high byte characters immediately
+        before occurrences of >.
+
+        This filter can be applied to all scripts which contain both English
+        characters and extended ASCII characters, but is currently only used by
+        ``Latin1Prober``.
+        """
+        filtered = BytesIO()
+        in_tag = False
+        prev = 0
+
+        for curr in range(len(buf)):
+            buf_char = buf[curr:curr + 1]
```

Comment:
For people unfamiliar with why this is necessary we should put a short comment on the fact this is necessary for byte"strings" on Py3 more than anywhere else.